omero/developers/Web/LinkingFromWebclient.txt

```diff
@@ -0,0 +1,188 @@
+using :property:`omero.web.open_with`.
+
+For example:
+
+ - Open images in a custom viewer
+ - Add images to OMERO.figure
+ - Link to external resources, E.g. open Dataset named '000397' with url https://www.ncbi.nlm.nih.gov/protein/000397
+
+Currently, 'Open With' options are shown in the context menu of the left-panel tree
+and are therefore only available for objects shown in the tree.
+
+Open image in viewer
+^^^^^^^^^^^^^^^^^^^^
+
+In the simplest case the minimum needed to add a viewer is "label" and "url_name".
+
+::
+
+    $ bin/omero config append omero.web.open_with '["My viewer", "url_name"]'
+
+We use ``reverse(url_name)`` to resolve a url from the url_name. If the url_name
+is not recognised (E.g. for external urls) the ``url_name`` will be used directly.
+
+By default, the ``Open With > My viewer`` menu option will only be enabled when a
+single image is selected. When the user clicks the image, a new popup window will be opened
+with the specified url, passing the image ID in the query string:
+
+::
+
+    url?image=:id
+
+Further parameters can be specified in an options object, as described below.
+
+Open in new tab
+^^^^^^^^^^^^^^^
+
+If you wish to open in a new browser tab instead of a popup window, you can add a ``target``
+attribute to the options:
+
+::
+
+    $ bin/omero config append omero.web.open_with '["My viewer", "url_name"], {"target":"_blank"}]'
+
+Supported objects
+^^^^^^^^^^^^^^^^^
+
+By default, 'Open with' options are enabled when only a single image is selected.
+This can be configured to support multiple images or other data types by adding a
+``supported_objects`` list to the options.
+Specify data types as E.g. ``project, dataset, image, screen, plate, acquisition``
+or use plurals to indicate that multiple objects are supported.
+E.g. ```images``` will enable the 'Open with' option when 1 or more images are selected.
+In the following example, we support a single ``dataset`` or multiple ``images``.
+
+::
+
+    $ bin/omero config append omero.web.open_with '["My viewer", "url_name"], {"supported_objects":["dataset", "images"]}]'
+
+JavaScript handlers
+^^^^^^^^^^^^^^^^^^^
+
+For complete control over handling of your 'Open with' action, or to specify
+more precise behaviour of the enabled status, you can write JavaScript functions
+that handle these.
+Add one or both of these function calls to a script, E.g. ```openwith.js```
+
+::
+    // Here we set an 'enabled' handler that is passed a list of selected
+    // objects and should return ``true`` if the 'Open with' option should
+    // be enabled.
+    // First argument is the label that we used above to identify the option
+    OME.setOpenWithEnabledHandler("My viewer", function(selected){
+        // selected is a list of objects containing id, name, type
+
+        // E.g. only support single objects
+        if (selected.length !== 1) return false;
+
+        // E.g. only support image with name ending in .svs
+        var obj = selected[0];
```

Comment:
@will-moore couldn't complain about docs.

I used them when testing the open with functionality, adding scripts and enabled easy enough, as well as url and options such as target so that only after I looked into the code for more analysis. I missed your mix-up in the code 'supported_objects' (which you fixed meanwhile) but that does not concern the documentation. You covered supported objects there.

So this is good to go.